lib/hl7/delimiters.rb

```diff
@@ -83,5 +91,33 @@ def unescape_replacements
         }
       )
     end
+
+    def escape_regexp
+      @escape_regexp ||= (
+        quoted = [
+          field,
+          component,
+          subcomponent,
+          repeat,
+          escape
+        ].compact.map { |str| Regexp.quote(str) }
+
+        Regexp.new(quoted.join("|"))
+      )
+    end
+
+    def escape_replacements
+      @escape_replacements ||= (
```

Comment:
Fine with me. I've been aware of this convention for a long time, but I never used to use it, because I was always in the habit of only ever _reading_ instance variables via an `attr_reader`. So, in effect, I treated _all_ instance variables as ones that I would never directly read, so there really was no important distinction between the two kinds. (At least, I don't think the point is to distinguish between instance variables that are used for memoizing. I think the point is to distinguish ones that you'll never read directly. That's why the same convention is used for local variables that you don't intend to use.)

Recently, though, I've started accessing instance variables directly. I'm not sure if that's good or not.

Anyhow, I'll take a look at the code in this library. If I make the change, I'm going to make it everywhere, in which case, it will be a separate PR.

Reply:
sounds good!